lib/internal/loader/ModuleRequest.js

```diff
@@ -35,22 +35,34 @@ loaders.set('esm', async (url) => {
 });
 
 // Strategy for loading a node-style CommonJS module
+// Uses babel and typescript style __esModule property for
+// attaching named imports due to the possiblity of `module.exports.default`
 loaders.set('cjs', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading CJSModule ${url}`);
-    const CJSModule = require('module');
-    const pathname = internalURLModule.getPathFromURL(new URL(url));
-    CJSModule._load(pathname);
+  debug(`Loading CJSModule ${url}`);
+  const CJSModule = require('module');
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const exports = CJSModule._load(pathname);
+  const es = !!exports.__esModule;
```

Comment:
I think we should promote the use of a `Symbol` instead. Adding a property might not be feasible for a lot of modules. We should also support `__esModule` for backward compat with babel.

Reply:
thats a fantastic idea, would attaching it to the `module` module, perhaps as `Module.esModule` be good?

Reply:
Perhaps something in the name to indicate it is a symbol for CommonJS interop cases? `esModuleInterop`?

Reply:
I'm fine with making it as long as we want. I would use `Symbol.for()` so it can be backward-compatibile with older versions of node that do not have the symbol, plus other tools that might want to process it.